main.go

```diff
@@ -245,13 +244,13 @@ func runWithLeaderElection(
 		// get elected before your background loop finished, violating
 		// the stated goal of the lease.
 		ReleaseOnCancel: true,
-		LeaseDuration:   leaderLeaseDuration,
-		RenewDeadline:   10 * time.Second,
+		LeaseDuration:   cfg.LeaseDuration,
+		RenewDeadline:   cfg.LeaseRenewDeadline,
 		RetryPeriod:     3 * time.Second,
 		WatchDog:        watchDog,
 		Callbacks: leaderelection.LeaderCallbacks{
 			OnStartedLeading: func(ctx context.Context) {
-				log.Infof("started leader: %s", id)
+				log.Infof("started leader: %s leaseDuration: %v, leaseRenewDeadline: %v", id, cfg.LeaseDuration, cfg.LeaseRenewDeadline)
```

Comment:
All looks good.
Please consider if this makes sense.
`log.WithFields("leaseDuration", strconv.Itoa(cfg.LeaseDuration), "leaseRenewDuration", strconv.Itoa(cfg.LeaseRenewDeadline)).Infof("started leader: %s", id)`

Reply:
done